analysis/R/assoc.R

```diff
@@ -0,0 +1,278 @@
+source("analysis/R/encode.R")
+source("analysis/R/decode.R")
+source("analysis/R/simulation.R")
+source("analysis/R/read_input.R")
+source("analysis/R/association.R")
+source("tests/gen_counts.R")
+
+# Analysis where second variable is basic RAPPOR
+TwoWayAlgBasic <- function(inp) {
+  ptm <- proc.time()
+  params <- ReadParameterFileMulti(inp$params)
+  stopifnot(inp$numvars == length(inp$maps) ||
+              inp$numvars == length(params))
+  map <- CorrectMapForAssoc(ReadMapFile(inp$maps[[1]], params = params[[1]]),
+                            params = params[[1]])
+  
+  # 2 way counts, marginal 1 counts, marginal 2 counts
+  counts <- lapply(1:3, function(i) ReadCountsFile(inp$counts[[i]]))
+  
+  # Prune candidates for variable 1
+  fit <- Decode(counts[[2]], map$rmap, params[[1]], quick = FALSE)$fit
+  print(fit)
+  found_strings <- fit[,"string"]
+  
+  if(length(found_strings) == 0)
+    stop("No strings found in 1-way marginal.")
+  
+  pruned_map <- lapply(map$map, function(z) z[,found_strings, drop = FALSE])
+  map2 <- lapply(1:length(pruned_map), function(z) {
+    m <- sparseMatrix(c(1), c(2), dims = c(1, 2))
+    colnames(m) <- c("FALSE", "TRUE")
+    m
+  })
+  # Final map
+  fmap <- CombineMaps(pruned_map, map2)$crmap
+  params2 <- params[[1]]
+  # Only for boolean
+  params2$k <- (params[[1]]$k) * 4
+  # Get true values (for debug purposes only)
+  truevals <- read.csv(inp$truevals)[,c("value1", "value2")]
+  truevals <- table(truevals) / sum(table(truevals))
+  colnames(truevals) <- c("FALSE", "TRUE")
+  marginal <- Decode2Way(counts[[1]], fmap, params2,
+                         fit = fit[,c("string", "proportion")])$fit
+  rs <- rowSums(truevals)
+  fits <- data.frame(string = rownames(as.data.frame(rs)),
+                     proportion = as.vector(rs))
+  truecol = NULL
+  for (rows in found_strings) {
+    for (cols in c("FALSE", "TRUE")) {
+      truecol <- c(truecol, truevals[rows, cols])
+    }
+  }
+  marginal <- cbind(marginal, true = truecol)
+  for (i in -3:3) {
+    fits_t <- fits
+    fits_t[,"proportion"] <- fits_t[,"proportion"] * (1 + i/10)
+    marginal <- cbind(marginal,
+                      more = Decode2Way(counts[[1]],
+                                 fmap,
+                                 params2,
+                                 fit = fits_t)$fit[,"Estimate"])
+    print("ABS")
+    print(0.5 * sum(abs(marginal$true-marginal[,i + 8])))
+  }
+  ed <- matrix(0, nrow = length(found_strings), ncol = 2)
+  colnames(ed) <- c("FALSE", "TRUE")
+  rownames(ed) <- found_strings
+  for (rows in rownames(ed)) {
+    for (cols in colnames(ed)) {
+      ed[rows, cols] <- marginal[paste(rows, cols, sep = "x"), "Estimate"] 
+    }
+  }
+
+  print(marginal)
+  print(sum(marginal[,"Estimate"]))
+  ed[is.na(ed)] <- 0
+  ed[ed<0] <- 0
+  
+  time_taken <- proc.time() - ptm
+  print("Two Way Algorithm Results")
+  # print(ed)
+  # print(ed[order(-rowSums(ed)), order(-colSums(ed))])
+}
+
+TwoWayAlg <- function(inp) {
```

Comment:
Move TwoWayAlg and TwoWayAlgBasic into another file that's not in this pull request.